readthedocs/vcs_support/backends/git.py

```diff
@@ -69,22 +79,24 @@ def checkout_revision(self, revision=None):
             branch = self.default_branch or self.fallback_branch
             revision = 'origin/%s' % branch
 
-        code, out, err = self.run(
-            'git', 'checkout', '--force', revision)
+        code, out, err = self.run('git', 'checkout', '--force', revision)
         if code != 0:
-            log.warning("Failed to checkout revision '%s': %s",
-                        revision, code)
+            log.warning("Failed to checkout revision '%s': %s", revision, code)
         return [code, out, err]
 
     def clone(self):
-        code, _, _ = self.run(
-            'git', 'clone', '--recursive', self.repo_url, '.')
+        code, _, _ = self.run('git', 'clone', '--recursive', self.repo_url, '.')
```

Comment:
Should this also be separated? I'm mean deleting the `--recursive` option https://git-scm.com/docs/git-clone/2.8.0#git-clone---recursive and make the submodules init/update on other step.

Reply:
Also that option was doped and now only `--recurse-submodules` is used on the latest git version https://git-scm.com/docs/git-clone#git-clone---recurse-submodulesltpathspec

Reply:
I think @stsewd is right here. We should remove the `--recursive` option. Otherwise, we will be trying to clone the submodules in the initial clone.

Reply:
Oh hrm, yeah. Good point!